tools/loris_db_upgrade.php

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/php
+<?php declare(strict_types=1);
+
+/**
+ * This script will upgrade the loris database to a specified release version.
+ * Usage: php loris_db_upgrade.php 21.0 ()      
+ * Errorlog: log/*.log
+ * PHP Version 7
+ *
+ * @category Install
+ * @package  Loris
+ * @author   Shen Wang <shen.wang2@mcgill.ca>
+ * @license  Loris license
+ * @link     https://www.github.com/aces/Loris/
+ * @package core
+ */
+require_once __DIR__ . "/generic_includes.php";
+ set_include_path(get_include_path().":../project/libraries:../php/libraries:");
+$path    = __DIR__ . "/../SQL/Release_patches";
+
+$files = array_diff(scandir($path), array('.', '..'));
+
+echo "This script will upgrade the LORIS database to the latest release version.\n";
+
+if (!file_exists('../project/config.xml')) {
+    exit("config.xml file doesn't exist. Aborting update.");
+}
+
+$v = $argv[1];
+if (floatval($v) < 14.0 || floatval($v) > 99.0 )
+ {
+   echo "Please input an available version!";
+   exit;
+ }
+
+
+$sql = null;
+$query = null;
+foreach($files as $file){
+
+    if (substr($file, 0, strpos($file,'_'))>=$v) {
+     print_r($file);
+      $a = file_get_contents(__DIR__ . "/../SQL/Release_patches/".$file);
+      $sql .= $a;
+
+    }
+}
+$lines = explode(PHP_EOL, $sql);
+foreach ($lines as $line) {
+  // remove comment line from sql file
+   if (substr($line, 0, 2) !== '--') {
+     $query .= $line;
+   }
+
+}
+
+$factory      = \NDB_Factory::singleton();
+$db_config    = $factory->config()->getSetting('database');
+
+
+$conn = mysqli_connect($db_config['host'], $db_config['adminUser'], $db_config['adminPassword'], $db_config['database']);
+
+if (!$conn) {
+    echo "Error: Unable to connect to MySQL." . PHP_EOL;
+    echo "Debugging errno: " . mysqli_connect_errno() . PHP_EOL;
+    echo "Debugging error: " . mysqli_connect_error() . PHP_EOL;
+    exit;
+}
+
+if (!mysqli_query($conn,$query)) {
+   $error = (mysqli_error_list($conn));
+   print_r($error);
+   write_log(date("h:i:sa"));
+   foreach ($error as $e) {
+       write_log("Error No.  => " .$e['errno']);
+       write_log("SQL State  => " .$e['sqlstate']);
+       write_log("Error      => " .$e['error']);
+   }
+  echo "Check error log under log directory!";
+} else {
+  echo "Upgrade completed successfully!";
+}
+
+function write_log($log_msg)
+{
+    $log_filename = "log";
+    if (!file_exists($log_filename))
+    {
+        // create directory/folder uploads.
+        mkdir($log_filename, 0777, true);
```

Comment:
```suggestion
        mkdir($log_filename, 0640, true);
```
This file should allow read and write ownership by the owner but nobody else should be able to modify it.